virtool/samples/oas.py

```diff
@@ -0,0 +1,469 @@
+                "is_legacy": False,
+                "isolate": "",
+                "labels": [],
+                "library_type": "normal",
+                "locale": "",
+                "name": "HX8",
+                "notes": "",
+                "nuvs": False,
+                "paired": True,
+                "pathoscope": True,
+                "quality": {
+                    "bases": [
+                        [36.0, 37.0, 37.0, 37.0, 37.0, 37.0],
+                        [36.0, 37.0, 37.0, 37.0, 37.0, 37.0],
+                    ],
+                    "composition": [
+                        [29.0, 18.0, 15.0, 36.5],
+                        [25.5, 19.0, 31.5, 22.0],
+                    ],
+                    "count": 94601674,
+                    "encoding": "Sanger / Illumina 1.9\n",
+                    "gc": 43.0,
+                    "length": [150, 150],
+                    "sequences": [
+                        0,
+                        0,
+                        0,
+                        0,
+                        0,
+                        0,
+                        0,
+                        0,
+                        0,
+                        0,
+                        0,
+                        18,
+                        298,
+                    ],
+                },
+                "reads": [
+                    {
+                        "download_url": "/samples/9zn468u9/reads/reads_1.fq.gz",
+                        "id": 713,
+                        "name": "reads_1.fq.gz",
+                        "name_on_disk": "reads_1.fq.gz",
+                        "sample": "9zn468u9",
+                        "size": 3540467819,
+                        "upload": None,
+                        "uploaded_at": "2022-05-21T00:10:27.418000Z",
+                    },
+                    {
+                        "download_url": "/samples/9zn468u9/reads/reads_2.fq.gz",
+                        "id": 714,
+                        "name": "reads_2.fq.gz",
+                        "name_on_disk": "reads_2.fq.gz",
+                        "sample": "9zn468u9",
+                        "size": 3321721014,
+                        "upload": None,
+                        "uploaded_at": "2022-05-21T00:11:10.743000Z",
+                    },
+                ],
+                "ready": True,
+                "subtractions": [{"id": "0nhpi36p", "name": "Malus domestica"}],
+                "user": {"administrator": True, "handle": "mrott", "id": "ihvze2u9"},
+            }
+        }
+
+
+class EditSampleSchema(BaseModel):
+    name: Optional[constr(strip_whitespace=True, min_length=1)] = None
+    host: Optional[constr(strip_whitespace=True)] = None
+    isolate: Optional[constr(strip_whitespace=True)] = None
+    locale: Optional[constr(strip_whitespace=True)] = None
+    notes: Optional[constr(strip_whitespace=True)] = None
+    labels: Optional[list] = None
+    subtractions: Optional[list] = None
```

Comment:
No default values. Remember to use `exclude_unset`. This will be compatible with the existing logic:
```py
if "labels" in data:
  non_existent_labels = await check_labels(pg, data["labels"])

  if non_existent_labels:
    return bad_labels_response(non_existent_labels)
```
Data will just be `dict` you get from calling `EditSampleSchema.dict(exclude_unset=True)` on the request data object.